zrml/rikiddo/src/types.rs

```diff
@@ -206,17 +219,26 @@ impl<F: Fixed, N: TryFrom<u128>> FromFixedToDecimal<F> for N {
     /// Craft a fixed point decimal number from a `Fixed` type (e.g. `Fixed` -> `Balance`).
     fn from_fixed_to_fixed_decimal(fixed: F, places: u8) -> Result<N, &'static str> {
         if places == 0 {
-            let mut result = fixed.to_num::<u128>();
+            let mut result = fixed
+                .checked_to_num::<u128>()
+                .ok_or("The fixed point number does not fit into a u128")?;
 
             // Arithmetic rounding (+1 if >= 0.5)
             if F::frac_nbits() > 0 {
                 if let Some(two) = F::checked_from_num(2) {
-                    // `from_num(1)` cannot panic if `from_num(2)` succeeded
-                    if let Some(res) = F::from_num(1).checked_div(two) {
-                        if fixed.frac() >= res {
-                            result = result.saturating_add(1);
+                    if let Some(one) = F::checked_from_num(1) {
+                        if let Some(res) = one.checked_div(two) {
+                            if fixed.frac() >= res {
+                                result = result.saturating_add(1);
+                            }
+                        } else {
+                            return Err("Unexpected overflow when dividing one by two");
                         }
+                    } else {
+                        return Err("Unexpectedly failed to convert 1 to fixed point number");
                     }
+                } else {
+                    return Err("Unexpectedly failed to convert 2 to fixed point number");
                 }
             }
```

Comment:
[nitpicking] We could formulate that a little more concise. With so many else branches, it requires some effort to assign them to the operation that caused it

```rust
if F::frac_nbits() > 0 {
    let one = F::checked_from_num(1)
        .ok_or("Unexpectedly failed to convert 1 to fixed point number")?;
    let two = F::checked_from_num(2)
        .ok_or("Unexpectedly failed to convert 2 to fixed point number")?;
    let half =
        one.checked_div(two).ok_or("Unexpected overflow when dividing one by two")?;

    if fixed.frac() >= half {
        result = result.saturating_add(1);
    }
}
```

Reply:
I remember that you have asked a question related to that based on that code example. I hope my answer was not misleading. Both ways are fine imo, it's just when the code becomes lengthy and all branches are used, it can become a bit hard to read.

Reply:
No, the answer wasn't misleading at all. :) I used this approach to avoid changing the style, but I agree that the alternative is more readable.